### PR TITLE
[DO NOT MERGE] feat: add auth module for more than 20byte address

### PR DESCRIPTION
**Pull request by nowooj**
It is for PoC and there is a `feature/address32` branch target to version 47 of main.

### CODE REVIEW

File: x/auth/keeper/keeper.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+type AccountKeeper struct {
+	authkeeper.AccountKeeper
+	key sdk.StoreKey
+}
+
+func NewAccountKeeper(
+	cdc codec.BinaryCodec, key sdk.StoreKey, ps paramtypes.Subspace, proto func() authtypes.AccountI,
+	maccPerms map[string][]string,
```
Comment (yoosah) [inline x/auth/keeper/keeper.go:19]: I think it needs to apply v0.47.x changes like bech32 and authority.

File: x/auth/keeper/account.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	iterator := sdk.KVStorePrefixIterator(store, types.AddressStoreKey(addr))
+	defer iterator.Close()
+	if !iterator.Valid() {
+		return nil
+	}
+
+	return ak.decodeAccount(iterator.Value())
+}
+
+func (ak AccountKeeper) decodeAccount(bz []byte) types.AccountI {
```
Comment (yoosah) [inline x/auth/keeper/account.go:20]: It seems better to move this to keeper.go like cosmos sdk's auth.

File: x/auth/keeper/keeper.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+)
+
+type AccountKeeper struct {
+	authkeeper.AccountKeeper
+	key sdk.StoreKey
```
Comment (yoosah) [inline x/auth/keeper/keeper.go:14]: Why does this need to be separate?

**Comment by nowooj** [comment]
Deprecated cosmos-sdk v0.45

**Pull request closed without merging**